lib/events/encryptedevent.cpp

```diff
@@ -30,3 +31,24 @@ EncryptedEvent::EncryptedEvent(const QJsonObject& obj)
 {
     qCDebug(E2EE) << "Encrypted event from" << senderId();
 }
+
+RoomEventPtr EncryptedEvent::createDecrypted(const QString &decrypted) const
+{
+    auto eventObject = QJsonDocument::fromJson(decrypted.toUtf8()).object();
+    eventObject["event_id"] = id();
+    eventObject["sender"] = senderId();
+    eventObject["origin_server_ts"] = originTimestamp().toMSecsSinceEpoch();
+    if(contentJson().contains("m.relates_to")) {
+        auto relates = contentJson()["m.relates_to"].toObject();
+        auto content = eventObject["content"].toObject();
+        content["m.relates_to"] = relates;
```

Comment:
Requires the newest `dev`.
```suggestion
    if (const auto relatesToJson = contentPart("m.relates_to"_ls); !relatesToJson.isUndefined()) {
        auto content = eventObject["content"].toObject();
        content["m.relates_to"] = relatesToJson.toObject();
```
I'm also not 100% sure that `content` does not end up detached from `eventObject` after calling `toObject()`. Did you check that it really works?

Reply:
That's why I'm adding it back in the next line